src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricExporter.cs

```diff
@@ -106,19 +122,17 @@ public override ExportResult Export(in Batch<Metric> metrics)
 
                 try
                 {
-                    wire_size += request.CalculateSize();
-
-                    // [abeaulieu] This need to be refactored since we will need several requests now.
-                    // request.AddMetrics(this.ProcessResource, metrics);
-                    // [abeaulieu] TODO: Spread the request load?
-                    if (!this.exportClient.SendExportRequest(request))
+                    if (request == null || !this.exportClient.SendExportRequest(request))
```

Comment:
When `request == null` here, we are done exporting all metrics, because there were no messages left. That is not a failure, so we shouldn't return failure. 

Because of this change, we no longer need `done`, and the while condition can be `request != null` instead.